### PR TITLE
Ifaa 9 12

### PR DESCRIPTION
**Pull request by asiatqc**
1 thing I haven't did - redo getByRoleName(String roleName)  with Role roleName.
I have just tired of dao. But all other things were resolved.
So, review please.

### CODE REVIEW

File: src/test/java/com/softserveinc/edu/ita/domains/User.java [hunk 1]
```diff
@@ -78,6 +80,22 @@ public void setCustomerType(String customerType) {
         this.customerType = customerType;
     }
 
+    public String getIsUserActive() {
+        return isUserActive;
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/domains/User.java:84]: is - only for booleans
getStatus() etc

File: src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java [hunk 2]
```diff
@@ -1,10 +1,58 @@
...
     public AdministrationPage(WebDriver driver) {
         super(driver);
     }
+
+    public NewUserPage clickCreateUserLink() {
+        driver.findElement(AdministrationPageLocators.CREATE_NEW_USER_LINK.getBy()).click();
+        return new NewUserPage(driver);
+    }
+
+    public void acceptAlert() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java:22]: palka - we talked about it - move acceptAlert() to base class

File: src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java [hunk 3]
```diff
@@ -1,10 +1,58 @@
...
+        int randomLoginRow = randomGenerator.nextInt(4) + 1;
+
+        WebElement loginCell = driver.findElement(By.xpath(
+                String.format(AdministrationPageLocators.LOGIN_CELL, randomLoginRow)));
+
+        return loginCell.getText();
+    }
+
+    public void clickLastButton() {
+        driver.findElement(AdministrationPageLocators.LAST_BUTTON.getBy()).click();
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java:40]: there's a click() method in base page object page, use it

File: src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java [hunk 4]
```diff
@@ -1,10 +1,58 @@
...
+        String login = driver.findElement(By.xpath(String
+                .format(AdministrationPageLocators.LOGIN_CELL, tableSize))).getText();
+        return login;
+    }
+
+    public void clickDeleteLastUser() {
+        WebElement table = driver.findElement(AdministrationPageLocators.USERS_TABLE.getBy());
+        int tableSize = table.findElements(AdministrationPageLocators.USERS_TABLE_ROWS.getBy()).size();
+        driver.findElement(By.xpath(String
+                .format(AdministrationPageLocators.DELETE_LINK, tableSize))).click();
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java:55]: use base page click()

File: src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    public AdministrationPage clickCreateButton() {
+        driver.findElement(NewUserPageLocators.CREATE_BUTTON.getBy()).click();
+        return new AdministrationPage(driver);
+    }
+
+    public void fillUserDataInput(By inputLocator, String inputValue) {
+        WebElement input = driver.findElement(inputLocator);
+        input.sendKeys(inputValue);
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java:28]: use base page click()

File: src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    public NewUserPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public AdministrationPage clickCreateButton() {
+        driver.findElement(NewUserPageLocators.CREATE_BUTTON.getBy()).click();
+        return new AdministrationPage(driver);
+    }
+
+    public void fillUserDataInput(By inputLocator, String inputValue) {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java:26]: why is this still needed ?

File: src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        WebElement input = driver.findElement(inputLocator);
+        input.sendKeys(inputValue);
+    }
+
+    public void fillLoginNameField(String loginName) {
+        WebElement input = driver.findElement(NewUserPageLocators.LOGIN_NAME_INPUT.getBy());
+        input.sendKeys(loginName);
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java:33]: why create webelement?
base page sendKeys()
same goes for all methods

File: src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+        WebElement input = driver.findElement(NewUserPageLocators.CONFIRM_PASSWORD_INPUT.getBy());
+        input.sendKeys(password + "_");
+    }
+
+    public void fillEmailField(String email) {
+        WebElement input = driver.findElement(NewUserPageLocators.EMAIL_INPUT.getBy());
+        input.sendKeys(email);
+    }
+
+    public void acceptAlert() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java:61]: move out to base page

File: src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+        Alert alert = driver.switchTo().alert();
+        alert.accept();
+    }
+
+    public void fillAllFields(User newUser) {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java:66]: use you methods here, no need for this
fillFirstNameField()
fillLastNameField()
etc

File: src/test/java/com/softserveinc/edu/ita/tests/CreateUserTest.java [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+        AdministrationPage administrationPage = userInfoPage.clickAdministrationTab();
+        final NewUserPage newUserPage = administrationPage.clickCreateUserLink();
+
+        newUserPage.fillAllFields(newUser);
+        // TODO try to resolve for valid and not valid data ?
+        administrationPage = newUserPage.clickCreateButton();
+
+        final User lastUser = DBUtility.getLastUser();
+        loggingAssert.assertEquals(newUser.getLogin(), lastUser.getLogin(),
+                "New user has stored into database.");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/tests/CreateUserTest.java:37]: assert message: "New user adding"

File: src/test/java/com/softserveinc/edu/ita/tests/CreateUserTest.java [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+        final User admin = DBUtility.getAdmin();
+        final UserInfoPage userInfoPage = homePage.logIn(admin.getLogin(), admin.getPassword());
+        final AdministrationPage administrationPage = userInfoPage.clickAdministrationTab();
+        final NewUserPage newUserPage = administrationPage.clickCreateUserLink();
+
+        newUserPage.clickCreateButton();
+        newUserPage.acceptAlert();
+
+        // test fields returns error message for empty data
+        loggingAssert.assertTrue(newUserPage.isElementDisplayed(NewUserPageLocators.LOGIN_NAME_ERROR_LABEL.getBy()),
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/tests/CreateUserTest.java:58]: use base page isElementDisplayed
same goes for all methods

File: src/test/java/com/softserveinc/edu/ita/utils/StringsGenerator.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    /**
+     * Random String generator
+     *
+     * @param symbolsSet set of symbols for generate string in valid_symbols.prop
+     * @param minimalLength      minimal length of returned string
+     * @param maximalLength      maximal length of returned string
+     */
+    public static String generateString(String symbolsSet, int minimalLength, int maximalLength) {
+
+        //final String SYMBOLS_FILENAME = "symbol_sets.properties";
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/utils/StringsGenerator.java:19]: palka

File: src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java (hunk 2 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java:22]: Protest. We talked to move if someone needs it, i've asked people in the morning, nobody needs.

File: src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java (hunk 6 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java:26]: Don't understand. If you mean clickCreateButton I can say that's old code. Now it's named
clickCreateUserLink

File: src/test/java/com/softserveinc/edu/ita/tests/CreateUserTest.java (hunk 11 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/tests/CreateUserTest.java:58]: fixed

File: src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+        super(driver);
+    }
+
+    public AdministrationPage clickCreateButton() {
+        driver.findElement(NewUserPageLocators.CREATE_BUTTON.getBy()).click();
+        return new AdministrationPage(driver);
+    }
+
+    public void fillUserDataInput(By inputLocator, String inputValue) {
+        WebElement input = driver.findElement(inputLocator);
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java:27]: why is this still needed - fillUserDataInput() ?

File: src/test/java/com/softserveinc/edu/ita/domains/User.java (hunk 1 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/domains/User.java:84]: fixed

File: src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java (hunk 2 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java:22]: fixed

File: src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java (hunk 3 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/page_object/AdministrationPage.java:40]: all clicks fixed

File: src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java (hunk 7 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java:33]: fixed

File: src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java (hunk 9 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/page_object/NewUserPage.java:66]: fixed

File: src/test/java/com/softserveinc/edu/ita/tests/CreateUserTest.java (hunk 10 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/tests/CreateUserTest.java:37]: fixed

File: src/test/java/com/softserveinc/edu/ita/utils/StringsGenerator.java (hunk 12 above)
Comment (asiatqc) [inline src/test/java/com/softserveinc/edu/ita/utils/StringsGenerator.java:19]: fixed

**Pull request closed without merging**